test/e2e/dns.go

```diff
@@ -47,6 +51,10 @@ func createDNSPod(namespace, wheezyProbeCmd, jessieProbeCmd string) *api.Pod {
 		ObjectMeta: api.ObjectMeta{
 			Name:      "dns-test-" + string(util.NewUUID()),
 			Namespace: namespace,
+			Annotations: {
+				"hostName": DNSPodHostName,
```

Comment:
These annotations have to be namespaced.  `net.beta.kubernetes.io/hostname` and `net.beta.kubernetes.io/hostname-set` ?